Make sure all failures to load module interfaces are diagnosed

jrose-apple opened the pull request:
...specifically, diagnosed in the parent DiagnosticEngine. This not only provides a better user experience, but makes sure that the compiler exits with a nonzero exit code even if the module goes unused.

rdar://problem/50789839

jrose-apple commented:
@swift-ci Please smoke test

harlanhaskins reviewed (approved):
Thank you!

nathawes commented on lib/Frontend/ParseableInterfaceModuleLoader.cpp:
```diff
@@ -564,6 +565,17 @@ class swift::ParseableInterfaceBuilder {
+      SWIFT_DEFER {
+        // Make sure to emit a generic top-level error if a module fails to
+        // load. This is not only good for users; it also makes sure that we've
+        // emitted in error in the parent diagnostic engine, which is what
```
typo: in error in -> an error in

nathawes reviewed (approved):
LGTM!

jrose-apple committed:
Make sure all failures to load module interfaces are diagnosed

...specifically, diagnosed in the parent DiagnosticEngine. This not
only provides a better user experience, but makes sure that the
compiler exits with a nonzero exit code even if the module goes
unused.

rdar://problem/50789839

jrose-apple commented:
@swift-ci Please smoke test
